x-pack/filebeat/input/gcs/job/helper.go

```diff
@@ -0,0 +1,114 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build !aix
+// +build !aix
+
+package job
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/elastic/beats/v7/libbeat/common/jsontransform"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+)
+
+var (
+	errBodyEmpty       = errors.New("body cannot be empty")
+	errUnsupportedType = errors.New("only JSON objects are accepted")
+)
+
+func httpReadJSON(body io.Reader) (objs []mapstr.M, rawMessages []json.RawMessage, status int, err error) {
+	if body == http.NoBody {
+		return nil, nil, http.StatusNotAcceptable, errBodyEmpty
+	}
+	obj, rawMessage, err := decodeJSON(body)
+	if err != nil {
+		return nil, nil, http.StatusBadRequest, err
+	}
+	return obj, rawMessage, http.StatusOK, err
+}
+
+func decodeJSON(body io.Reader) (objs []mapstr.M, rawMessages []json.RawMessage, err error) {
```

Comment:
I don't think the names add anything here.
```suggestion
func decodeJSON(body io.Reader) ([]mapstr.M, []json.RawMessage, error) {
```